### DISCUSSION TITLE
How Many Hidden Layers to Use in Leela Chess Zero?

### DISCUSSION THREAD

**Discussion by ginkgo20** [LeelaChessZero/lc0#1887] [Q&A] [answered]
I want to know about Leela architecture. Is there a huge difference between Alpha0 & Leela0  in architecture?

**Comment by Vladislav3107** [comment]
where is my money=)

**Reply by ginkgo20** [reply]
What?!

**Comment by mooskagh** [comment, marked as answer]
Initially, the topology of the Lc0 network was exactly the same as a0.

However, with time, some changes were introduced. Particularly,

* Number of residual blocks changed over time, and the most recent resnet-based nets have 15 blocks (of 2 convolutional layers each).
* Lc0 uses Squeeze-and-Excitation blocks instead of pure resnet.
* Policy head is convolutional now rather than fully connected.
* Value head returns W,D,L rather than a single V value.
* Moves-left-head was not present in the a0 paper at all.

Moreover, the most recent and strongest networks are not resnet-based at all, they are [transformer based](https://github.com/Ergodice/lczero-training/blob/attention-net-body/README.md).
